scripts/invitejoin.pl

```diff
@@ -33,41 +33,213 @@
     description => 'This script will join a channel if somebody invites you to it.',
     license     => 'Public Domain',
     url         => 'http://irssi.hauwaerts.be/invitejoin.pl',
-    changed     => 'Sun Apr 11 12:38:18 2004',
+    changed     => 'Di 3. Jan 19:46:51 CET 2017',
 );
 
-## Comments and remarks.
-#
-# This script uses settings.
-# Use /SET to change the value or /TOGGLE to switch it on or off.
-#
-#    Setting:     invitejoin
-#    Description: If this setting is turned on, you will join the channel
-#                 when invite to.
-#
-##
+my $help = <<EOF;
+
+/SET    invitejoin 0|1
+/TOGGLE invitejoin
+          Description: If this setting is turned on, you will join the channel
+          when invited to.
+
+Default is to follow every invite, you can specify a list of allowed nicks.
+
+/INVITEJOIN [addnick <ircnet> <nick>]
+            [delnick <ircnet> <nick>]
+            [listnick]
+            [help]
 
+addnick:     Add a new nickname on the given net as allowed autoinvite source.
+delnick:     Delete a nickname from the allowed list.
+listnick:    Display the contents of the allowed nickname list.
+help:        Display this useful little helptext.
+
+Examples: (all on one line)
+/INVITEJOIN addnick Freenode ChanServ
+
+Note: This script doesn't allow wildcards
+EOF
+
+my @allowed_nicks = ();
+my $allowed_nicks_file = "invitejoin.nicks";
+
+my $irssidir = Irssi::get_irssi_dir();
 
 Irssi::theme_register([
+    'invitejoin_usage', '%R>>%n %_Invitejoin:%_ Insufficient parameters: Use "%_/INVITEJOIN help%_" for further instructions.',
+    'invitejoin_help', '$0',
     'invitejoin_loaded', '%R>>%n %_Scriptinfo:%_ Loaded $0 version $1 by $2.',
-    'invitejoin_invited', '%R>>%n %_Invitejoin:%_ Joined $1 (Invited by $0).'
+    'invitejoin_invited', '%R>>%n %_Invitejoin:%_ Joined $1 (Invited by $0).',
+    'invitejoin_usage_add_nick', '%R>>%n %_Invitejoin:%_ Insufficient parameters: Usage "%_/INVITEJOIN addnick ircnet ChanServ%_".',
+    'invitejoin_no_net', '%R>>%n %_Invitejoin:%_ Unknown Irssi ircnet %_$0%_.',
+    'saved_nick', '%R>>%n %_Invitejoin:%_ Added allowed nick "%_$1%_" on %_$0%_.',
+    'invitejoin_delusage', '%R>>%n %_Invitejoin:%_ Insufficient parameters: Usage "%_/INVITEJOIN delnick ircnet nick%_".',
+    'invitejoin_delled', '%R>>%n %_Invitejoin:%_ Deleted %_$1%_ on %_$0%_ from allowed list.',
+    'invitejoin_nfound', '%R>>%n %_Invitejoin:%_ The nick %_$1%_ on %_$0%_ could not be found.',
+    'allowed_nicks_info', '%_Ircnet             Nick%_',
+    'allowed_nicks_empty', '%R>>%n %_Invitejoin:%_ Your allowed nick list is empty.',
+    'allowed_nicks_print', '$[18]0 $1',
+    'invite_denied', '%R>>%n %_Invitejoin:%_ Invite from nick %_$1%_ on %_$0%_ to %_$2%_ not followed because it is not in the allowed list.',
 ]);
 
-sub invitejoin {
+sub load_allowed_nicks {
+    my ($file) = @_;
+    @allowed_nicks = ();
+    if (-e $file) {
+        local *F;
+        open(F, "<", $file);
```

Comment:
while you're in the process of modifying the script, why not replace this with "modern style" `open my $f, "<", $file`